### DISCUSSION TITLE
Typesafe config

### DISCUSSION THREAD

**Discussion by c0rdeiro** [rainbow-me/rainbowkit#1898] [Q&A] [answered]
Hey, I'm not sure how to achieve this [https://wagmi.sh/react/typescript#declaration-merging](https://wagmi.sh/react/typescript#declaration-merging) when using default config by Rainbowkit.
The objective is to get the typesafety (example with the available chains)

**Comment by c0rdeiro** [comment]
I think a solution more close to connectkit would allow this, where the `getDefaultConfig `returns the object to insert in wagmi's `createConfig`, so I can have the wagmi config on my side. (here are their [docs](https://docs.family.co/connectkit/getting-started#getting-started-section-3-implementation) )
```typescript
import { WagmiProvider, createConfig, http } from "wagmi";
import { mainnet } from "wagmi/chains";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ConnectKitProvider, getDefaultConfig } from "connectkit";

const config = createConfig(
  getDefaultConfig({
    // Your dApps chains
    chains: [mainnet],
    walletConnectProjectId: process.env.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID,
    appName: "Your App Name",
  }),
);
```

**Reply by 0xmkh** [reply]
We're aiming to release this fix very soon.

**Reply by c0rdeiro** [reply]
thanks!

**Reply by c0rdeiro** [reply]
is there an open issue/PR I can track?

**Reply by 0xmkh** [reply, marked as answer]
Yep. Check it out here: https://github.com/rainbow-me/rainbowkit/pull/1893
